src/metabase/lib/field.cljc

```diff
@@ -307,6 +307,12 @@
    ;; because the query processor doesn't produce nested display-names.
    {:display-name (lib.metadata.calculation/display-name query stage-number field-metadata)
     :long-display-name (lib.metadata.calculation/display-name query stage-number field-metadata :long)}
+   ;; Include description and fingerprint if they're present on the column. Only proper fields or columns from a model
+   ;; have these, not aggregations or expressions.
+   (when-let [description (:description field-metadata)]
+     {:description description})
+   (when-let [fingerprint (:fingerprint field-metadata)]
+     {:fingerprint fingerprint})
```

Comment:
It would be nice to have a test for this - will this map convert to js properly?

Reply:
It does, but it certainly doesn't hurt to test it. Doubly so as a couple of the keys are weird, like `"nil%"`.

Reply:
:sheep: Actually I found a bug by adding this test - the `displayInfo` keys are all converted to `camelCase` so while the `nil%` goes through intact, multi-word keys like `:distinct-count` become `"distinctCount"`, which doesn't match the fingerprint type used elsewhere.

I've updated the TS types accordingly.

PTAL, there's a fair chunk of new BE and FE code.